src/video/kmsdrm/SDL_kmsdrmvideo.c

```diff
@@ -1373,10 +1373,14 @@ KMSDRM_SetWindowGammaRamp(_THIS, SDL_Window * window, const Uint16 * ramp)
     SDL_VideoData *viddata = (SDL_VideoData*)windata->viddata;
     SDL_VideoDisplay *disp = SDL_GetDisplayForWindow(window);
     SDL_DisplayData* dispdata = (SDL_DisplayData*)disp->driverdata;
-    if (KMSDRM_drmModeCrtcSetGamma(viddata->drm_fd, dispdata->crtc->crtc_id, 256, &ramp[0*256], &ramp[1*256], &ramp[2*256]) == -1)
+    Uint16* tempRamp = SDL_calloc(3 * 2,256);
+    SDL_memcpy(tempRamp,ramp,3 * 2 * 256);
+    if (KMSDRM_drmModeCrtcSetGamma(viddata->drm_fd, dispdata->crtc->crtc_id, 256, &tempRamp[0*256], &tempRamp[1*256], &tempRamp[2*256]) == -1)
```

Comment:
I assume that drmModeCrtcSetGamma() takes non-const pointers? If the function really doesn't touch the inputs, you should be safe to do a const cast here instead of chancing a memory allocation.

Reply:
`drmModeCrtcSetGamma()` takes non-const pointers. And given the implementation-dependent nature of the Linux DRM graphics drivers as well as the scarcity of the libdrm documentation, it's better to avoid chancing modification of user data.